### PR TITLE
More generous binding timeouts in TcpSpec

### PR DESCRIPTION
**Pull request by johanandren**
Improves #21014

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 1]
```diff
@@ -430,7 +430,7 @@ class TcpSpec extends StreamSpec("akka.stream.materializer.subscription-timeout.
           .run()
 
       // make sure that the server has bound to the socket
-      val binding = Await.result(bindingFuture, 100.millis)
+      val binding = Await.result(bindingFuture, 1.second)
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: why not make it `remainingOrDefault`?

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: Mostly because I don't know where/how that's applicable/good to use, everywhere?

Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: * `remaining` is first choice (requires `within` block)
* `remainingOrDefault`is second choice
* `3.seconds` is third choice if not using testkit
* lower must come with a very good reason

Read up on `remaining` and friends in https://github.com/akka/akka/blob/master/akka-testkit/src/main/scala/akka/testkit/TestKit.scala

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: Thanks. Maybe we should create some kind of test-suite-best-practices doc?

**Comment by ktoso** [comment]
LGTM

**Pull request merged**